service/discovery/azure/key_vault.go

```diff
@@ -158,6 +191,15 @@ func (d *azureKeyVaultDiscovery) initKeysClient() (err error) {
 	d.clients.keysClient, err = initClient(d.clients.keysClient, d.azureDiscovery, armkeyvault.NewKeysClient)
 	return
 }
+func (d *azureKeyVaultDiscovery) initSecretsClient() (err error) {
+	d.clients.secretsClient, err = initClient(d.clients.secretsClient, d.azureDiscovery, armkeyvault.NewSecretsClient)
+	return
+}
+
+func (d *azureKeyVaultDiscovery) initCertificatesClient(kv *armkeyvault.Vault) (err error) {
+	d.clients.certificationsClient, err = azcertificates.NewClient(util.Deref(kv.Properties.VaultURI), d.cred, nil)
+	return
+}
```

Comment:
Just an idea for line 216: Shouldn't we discover also the keyVaults that are not used at the moment comparable to the virtual machines. We discover all VMs whether they are stopped or not. We could use an property "enabled" in the keyVault object.